lib/Doctrine/ORM/Query/ParameterTypeInferer.php

```diff
@@ -53,7 +53,7 @@ public static function inferType($value)
             return Type::BOOLEAN;
         }
 
-        if ($value instanceof \DateTime) {
+        if ($value instanceof \DateTime || $value instanceof \DateTimeImmutable) {
```

Comment:
Yes, undefined class in instanceof is not an error, it returns false. I think complicating the condition is unnecessary, since it's working in proposed form :-) http://3v4l.org/6KEnb